libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -190,7 +190,7 @@ void DeclarationTypeChecker::endVisit(UserDefinedTypeName const& _typeName)
 		m_errorReporter.fatalTypeError(
 			5172_error,
 			_typeName.location(),
-			"Name has to refer to a struct, enum or contract."
+			"Name has to refer to a user-defined type."
```

Comment:
This includes user-defined value types as well and initially in the operator PR we changed the message to:
```
Name has to refer to a user-defined value type, struct, enum or contract.
```
These are basically all possible user-defined types though so I think the generic version is better.

I still feel that this message could be better though. It appears when you use an identifier in a place where we expect a type. It seems to imply that the compiler expects a user-defined type but that's not necessarily true. It just means that **if** you're trying to use a custom name here, that name has to refer to a user-defined type. Does not mean you have to use an identifier though - you can use a built-in type as well in some cases.

Reply:
```suggestion
			"Name has to refer to a user-defined type or built-in type."
```
Would this be enough?

Reply:
I prefer the original message from the operators PR as it's more explicit, even I can get your point.

Reply:
Also, this would not be entirely correct. You may or may not be able to use a built-in type. We don't really know that in the context where we catch this error so we should not suggest that.

Reply:
Yeah, but it is either a user-defined or a built-in type, right? In that case, since we don't know the context, I think it would be correct to mention all the possibilities and let the user judge what should be the best.

Reply:
> Yeah, but it is either a user-defined or a built-in type, right? 

Honestly, I'm not sure and I'd rather not guess. Technically you can have syntax where either a type or something else is expected. The only thing we really know when we get this message is that we see some identifier and having an identifier in this place is legal but it has to be the name of a type. We don't know what else might be allowed here.